nox/manifest.py

```diff
@@ -282,6 +295,8 @@ def notify(self, session: Union[str, SessionRunner]) -> bool:
         # the end of the queue.
         for s in self._all_sessions:
             if s == session or s.name == session or session in s.signatures:
+                if posargs is not None:
+                    s.posargs = posargs
```

Comment:
I think I'll probably just cast it to a list in `Session.notify` since it'll make the internal API a little simpler if you can make some more strict assumptions about the type.